clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -3717,6 +3717,16 @@ def err_sme_definition_using_za_in_non_sme_target : Error<
   "function using ZA state requires 'sme'">;
 def err_sme_definition_using_zt0_in_non_sme2_target : Error<
   "function using ZT0 state requires 'sme2'">;
+def warn_sme_streaming_pass_return_vl_to_non_streaming : Warning<
+  "passing a VL-dependent argument to/from a function that has a different"
+  " streaming-mode, the streaming and non-streaming vector lengths may be"
+  " different">,
+  InGroup<AArch64SMEAttributes>;
```

Comment:
These warnings should be disabled by default, as the user may intentionally pass/return scalable vectors, knowing something about the target they're compiling for.